src/test/java/org/broadinstitute/hellbender/tools/HtsgetReaderIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+        IntegrationTestSpec.assertEqualTextFiles(output, expected, "@");
+    }
+
+    // Test ability to filter by class
+    @Test
+    public void testHeaderOnly() throws IOException {
+        final File expected = new File(getToolTestDataDir(), "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam.header");
+        final File output = createTempFile("output", ".bam");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .add("U", ENDPOINT)
+                .add("ID", "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam")
+                .add("class", "header")
+                .addOutput(output);
+
+        runCommandLine(args);
+        IntegrationTestSpec.assertEqualTextFiles(output, expected, "@");
+    }
+
+    // Test ability to filter by reference name
+    @Test
+    public void testReferenceName() throws IOException {
+        final File expected = new File(getToolTestDataDir(), "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam.refname");
+        final File output = createTempFile("output", ".bam");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .add("U", ENDPOINT)
+                .add("ID", "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam")
+                .add("reference-name", "chr1")
+                .addOutput(output);
+
+        runCommandLine(args);
+        IntegrationTestSpec.assertEqualTextFiles(output, expected, "@");
+    }
+
+    // Test ability to filter by field
+    @Test
+    public void testField() throws IOException {
+        final File expected = new File(getToolTestDataDir(), "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam.field");
+        final File output = createTempFile("output", ".bam");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .add("U", ENDPOINT)
+                .add("ID", "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam")
+                .add("field", "QNAME")
+                .addOutput(output);
+
+        runCommandLine(args);
+        IntegrationTestSpec.assertEqualTextFiles(output, expected, "@");
+    }
+
+    // Test ability to filter by start and end
+    @Test
+    public void testStartEnd() throws IOException {
+        final File expected = new File(getToolTestDataDir(), "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam.startend");
+        final File output = createTempFile("output", ".bam");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .add("U", ENDPOINT)
+                .add("ID", "A1-B000168-3_57_F-1-1_R2.mus.Aligned.out.sorted.bam")
+                .add("reference-name", "chr1")
+                .add("start", "1000")
+                .add("end", "2000")
+                .addOutput(output);
+
+        runCommandLine(args);
+        IntegrationTestSpec.assertEqualTextFiles(output, expected, "@");
+    }
+
+    @DataProvider(name = "invalidParams")
+    public Object[][] invalidParams(){
+        return new Object[][]{
```

Comment:
Is it valid to specify start but not end and viceversa?

Reply:
Start without end is valid but not vice versa. Start and end both require reference. This is taken care of by the switch to using a SimpleInterval since the spec behaves the same way.